l3build-arguments.lua

```diff
@@ -158,39 +158,38 @@ option_list =
 
 -- This is done as a function (rather than do ... end) as it allows early
 -- termination (break)
-local function argparse()
-  local result = { }
-  local names  = { }
-  local long_options =  { }
-  local short_options = { }
+A.argparse = function (arg)
+  local result = {}
+  local names  = {}
+  local long_options =  {}
+  local short_options = {}
   -- Turn long/short options into two lookup tables
-  for k,v in pairs(option_list) do
-    if v["short"] then
-      short_options[v["short"]] = k
+  for k,v in pairs(A.option_list) do
+    if v.short then
+      short_options[v.short] = k
     end
     long_options[k] = k
   end
-  local args = args
```

Comment:
I have a feeling this was deliberate: I'll check back to be on the safe side

Reply:
`args` is never used. maybe a typo ?
`local arg = arg` ?
If the purpose is to make a local safe copy, then it better be at the top of the module.

Reply:
Probably a typo from some earlier refactoring, then: I was just checking as there are some oddities that are deliberate. I wanted to be sure, that was all.

Reply:
> If the purpose is to make a local safe copy, then it better be at the top of the module.

Well, if it was intended as performance enhancement, then it is better inside of the function if it is only used inside of the function and the function is only called once, since then it doesn't become a upvalue but just a regular stack entry. (Of course that mostly matters when it's accessed *a lot* which isn't the case here)